js/src/components/adaptive-form/adaptive-form.js

```diff
@@ -56,9 +63,35 @@ function AdaptiveForm( { children, ...props }, ref ) {
 		...formRef.current,
 	} ) );
 
+	// Enhancements related.
+	// Additional enhancements to make AdaptiveForm have more useful or reusable features.
+	// It could also be the playground of the practical instances before pushing them upstream.
+	const isMounted = useIsMounted();
+	const [ submission, setSubmission ] = useState( [ false, false ] );
+	const [ isSubmitting, isSubmitted ] = submission;
+
+	if ( onSubmit ) {
+		props.onSubmit = async function ( values ) {
+			setSubmission( [ true, false ] );
+
+			await onSubmit.call( this, values, adapterRef.current.submitter );
+
+			if ( isMounted() ) {
+				setSubmission( [ false, true ] );
+				adapterRef.current.submitter = null;
+			}
+		};
+	}
```

Comment:
It's unclear for me what this chunk of code really does. 

- Why submission is a tuple of two values? What those values are doing?
- Why we are attaching the `submission` state to the `isSubmitting` state?

Reply:
Adjusted and added in https://github.com/woocommerce/google-listings-and-ads/pull/1794/commits/7ccbc8f8e5568a6f74bbd15d337fbae286ca05f6.

> What those values are doing?

I didn't add the description of these two state values since `AdaptiveFormContext` already has [their JSDocs](https://github.com/woocommerce/google-listings-and-ads/blob/7ccbc8f8e5568a6f74bbd15d337fbae286ca05f6/js/src/components/adaptive-form/adaptive-form-context.js#L18-L21). Let me know if you think it should still have copies of them here.

> * Why we are attaching the `submission` state to the `isSubmitting` state?

I'm not sure what this question means, could you help clarify?